hydra-main/src/main/java/com/addthis/hydra/minion/HostLocation.java

```diff
@@ -50,7 +55,62 @@ public String getPhysicalHost() {
         return physicalHost;
     }
 
+    /**
+     * Get the HostLocation value corresponding to the given AvailabilityDomain.
+     */
+    public String getPriorityAd(AvailabilityDomain ad) {
+        if (ad == AvailabilityDomain.DATACENTER) {
+            return this.getDataCenter();
+        } else if (ad == AvailabilityDomain.RACK) {
+            return this.getRack();
+        } else {
+            // HOST or NONE or anything else
+            return this.getPhysicalHost();
+        }
+    }
+
+    @Override
+    public int hashCode() {
+        return this.toString().hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || obj.getClass() != getClass()) {
+            return false;
+        }
+        HostLocation location = (HostLocation) obj;
+        return location.getDataCenter().equals(this.dataCenter) &&
+               location.getRack().equals(this.rack) &&
+               location.getPhysicalHost().equals(this.physicalHost);
+    }
+
     public String toString() {
         return "dataCenter=" + dataCenter + ", rack=" + rack + ", physicalHost=" + physicalHost;
     }
+
+    /**
+     * Returns a score of distance between this and another HostLocation.
+     * Zero means maximum distance.
+     * This method is used to sort HostLocations by their distance score.
+     */
+    public int assignScoreByHostLocation(HostLocation o) {
+        if (this.getDataCenter().equals(o.getDataCenter())) {
+            if (!this.getRack().equals(o.getRack())) {
+                // Same dataCenter, different rack
+                return AvailabilityDomain.RACK.score;
```

Comment:
don't understand how this works.. how are the returning score determined?

Reply:
Just picked a convention that lets us easily sort based on "AD-distance" score :)